contracts/BidEvaluation.sol

```diff
@@ -16,20 +18,29 @@ contract BidEvaluation {
 
     event BidEvaluated(uint256 indexed rfpId, address indexed vendor, bool isWinner);
 
-    constructor(address _bidSubmissionContractAddress) {
+    constructor(
+        address _bidSubmissionContractAddress, 
+        address _vendorRegistrationContractAddress
+    ) {
         bidSubmissionContract = BidSubmission(_bidSubmissionContractAddress);
+        vendorRegistrationContract = VendorRegistration(_vendorRegistrationContractAddress); 
     }
 
     // Function to evaluate bids for a specific RFP
     function evaluateBids(uint256 _rfpId) public {
         BidSubmission.Bid[] memory bids = bidSubmissionContract.getBidsForRFP(_rfpId);
 
         // Placeholder for evaluation logic
-        // In a real implementation, this would involve complex logic
-        // For simplicity, we are just selecting the first bid as the winner
-        if (bids.length > 0) {
-            winningBids[_rfpId] = bids[0].vendor;
-            emit BidEvaluated(_rfpId, bids[0].vendor, true);
+        // We check and find the first bid in the array that has an eligible vendor if not, we move on the the next
+        for (uint i = 0; i < bids.length; i++) {
+            // Check if each vendor is eligible before proceeding with evaluation
+            if (vendorRegistrationContract.isVendorEligible(bids[i].vendor)) {
+                // In a real implementation, this would involve complex logic
+                // For simplicity, we are just selecting the first bid that is eligible as the winner
```

Comment:
Hopefully we don't loose points because of this 😄

Reply:
Yeah, I hope so too. I cannot seem to find a way use a require statement that would check every single vendor for their eligibility. Unfortunately using a require statement in place of the if statement reverts the whole transaction at the first sign of an ineligible vendor and not check the entire list of vendors.